src/iris/ui/static/js/iris.js

```diff
@@ -2111,6 +2147,61 @@ iris = {
       });
       this.redrawApplicationDropdown();
     },
+    createCustomNotificationTable: function(){
+      var template = Handlebars.compile(this.data.customNotificationTemplate),
+          settings = this.data.settings,
+          self = this;
+      this.data.$customNotificationTable.empty();
+      this.data.$customNotificationTable.append(template(settings));
+      this.data.$customNotificationTable.find('button.delete-custom-app-button').each(function() {
+        $(this).click(function() {
+          self.deletePerCustomApp($(this));
+        });
+      });
+
+      this.data.$customNotificationTable.find('button.edit-custom-app-button').each(function() {
+        $(this).click(function() {
+          self.editPerCustomApp($(this));
+        });
+      });
+
+      this.redrawApplicationNotificationDropdown();
+    },
+    createCustomNotificationModal: function(app){
+
+      var template = Handlebars.compile(this.data.createCustomNotificationModalTemplate),
+              settings = this.data.settings,
+              customNotificationModal = this.data.$customNotificationModal;
+
+      // call out to the back end to get all categories for this app
+      var appNotificationCategories = [];
+      $.ajax({
+        type: "GET",
+        url: '/v0/categories/' + app,
+        success: function(resp)
+        {
+          appNotificationCategories = resp;
+          appNotificationCategories.forEach((appSetting, i) => {
```

Comment:
I'm seeing inconsistencies here as some parts/most of the Iris codebase uses ES5-style closures and this new code uses ES6 arrow functions.

I think a while ago Saif suggested all Iris code be ES5-compatible, as we're not using a transpiler like babel to convert new ES6 features like arrow functions to the ES5 equivalents.

On the other hand, all modern browsers support arrow functions, and if Iris used more of them the need for using variables named `self` for things like ajax callbacks would be obviated.

Reply:
I got rid of the arrow functions for consistency.